### ISSUE TITLE
Reduce CLS on segmented pages

### ISSUE THREAD

**Issue by sujato** [suttacentral/suttacentral#2360]
Apparently Google dislikes the fact that segmented pages are composed and thus have a relatively large CLS. It seems that Google is very strict on the CLS now and wants a CLS to be less than 0.1s to get a good score or 0.25 s to get "needs improvement".

![Screenshot-from-2022-01-10-13-13-58](https://user-images.githubusercontent.com/6112010/148891980-58a1ccae-3b86-4888-8b90-b02af5236b0d.png)

## possible remedies

- SSR
- CSS `content-visibility`
- fixed `height` on segments in dual-language views
- delay rendering footer (maybe use `content-visibility` here)

**Comment by sujato** [comment]
Testing `content-visibility` on Lighthouse (Chrome 98.0.4758.9 (Official Build) dev (64-bit))

**tl;dr: there's no advantage to using `content-visibility: auto`.**

Weird patterns: desktop times mostly go down, while mobile is suspiciously consistent.

using dn16

/dn16/en/sujato?layout=sidebyside&reference=none&notes=asterisk&highlight=false&script=latin

```
  #segmented_text_content{
    content-visibility: visible/auto
  }
  ```

content-visibility: auto

#### desktop

.424
.153
.036
.004
0
0

(try a second round, still warmed up)

.761
.643
.181
.08
.321

#### mobile

.662
.662
.662

### content-visibility: visible

#### desktop

.129
.055
0

(try a second round, still warmed up)

.075
.054

#### mobile

.662
.665
.662

**Comment by ihongda** [comment]
![Image](https://github.com/user-attachments/assets/46b13b2c-e8d0-42cb-9b0b-26fa4b02b3b8)

Bhante @sujato I optimized the code and delayed loading some components, which seemed to improve it a bit.
